mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -1377,6 +1377,38 @@ struct AMDGPUDPPLowering : public ConvertOpToLLVMPattern<DPPOp> {
   }
 };
 
+struct AMDGPUSwizzleBitModeLowering
+    : public ConvertOpToLLVMPattern<SwizzleBitModeOp> {
```

Comment:
unrelated to this PR, but I think it'd be more intuitive if the op name is`dsSwizzleOp`, and we can have the QDMode and the BitMode as a attribute of this op. Additionally we can also reference https://gpuopen.com/learn/amd-gcn-assembly-cross-lane-operations. :)

Reply:
QDMode and BitMode variants have a different offsets format, my idea was to have them as separate ops in `AMDGPU`, so user won't need to bother with offset bitpacking even if they are both lowered to `ds_swizzle` eventually.